src/Workspaces/Core/Portable/FindSymbols/SymbolFinder.cs

```diff
@@ -271,7 +272,15 @@ public static IEnumerable<TSymbol> FindSimilarSymbols<TSymbol>(TSymbol symbol, C
                 foreach (var linkedDocumentId in originalDocument.GetLinkedDocumentIds())
                 {
                     var linkedDocument = solution.GetRequiredDocument(linkedDocumentId);
+
+                    // It's possible for us to have a solution snapshot where only part of a linked set of documents has
+                    // been updated.  As such, the other linked docs may have different contents/sizes than the original
+                    // doc we started with.  Skip those files as there's no sensible way to say that we have linked
+                    // symbols here when the contents are not the same.
```

Comment:
❔ Can we take this one step further and check that the `SourceText` is the same? While the syntax trees might differ, I would expect linked documents to at least share the `SourceText` internally.

Reply:
i'd prefer to not accidentally cause us to break where someone legally updates the different forks with different source-texts :)

Reply:
Is there another way to do content validation efficiently? Or perhaps the number of cases of mismatched SourceText is so rare that we can do checksum comparison if the length is the same but the instance is different?

Reply:
generally speaking, i think this check, along with the check below (For name/kind) is sufficient.